src/grid/Grid.tsx

```diff
@@ -209,7 +240,7 @@ export function Grid<
   const [gridRoot, gridRef2] = useState<HTMLElement | null>(null);
   const gridRef = useMergedRefs<HTMLElement>(gridRef1, gridRef2);
 
-  const [layoutRoot, setLayoutRoot] = useState<HTMLElement | null>(null);
+  const [layoutRoot, layoutRef] = useState<HTMLElement | null>(null);
```

Comment:
This is either wrong or there is some trick how forwardRef works under the hood. does it just call the Ref with the content so we can also use a state ?

Reply:
Oh right, this is making use of React's [callback ref](https://react.dev/reference/react-dom/components/common#ref-callback) feature, which allows any callback of type `(value: R) => void` to act as a `Ref<R>`. Since `setLayoutRoot` has type `(value: HTMLElement | null) => void`, it can be used as a `Ref<HTMLElement | null>`.

Would you find this clearer if the name was still `setLayoutRoot` instead of `layoutRef`?

Reply:
It would be nice if they have a naming convention for this. (sth like a setRef)
But knowing this is enough so the name makes sense.